kafka/source/pkg/reconciler/kafkasource.go

```diff
@@ -206,6 +219,188 @@ func checkResourcesStatus(src *v1alpha1.KafkaSource) error {
 	return nil
 }
 
+func (r *Reconciler) scaleKafkaSource(ctx context.Context, ra *v1.Deployment, src *v1alpha1.KafkaSource) {
+	r.readScalingAnnotations(ctx, src)
+	// no scaling annotatins so no scaling work needed
+	if r.minReplicaCount == nil && r.maxReplicaCount == nil {
+		return
+	}
+	_, error := r.deployKedaScaledObject(ctx, ra, src)
+	if error != nil {
+		// additional logging?
+	}
+}
+
+func (r *Reconciler) deployKedaScaledObject(ctx context.Context, ra *v1.Deployment, src *v1alpha1.KafkaSource) (*unstructured.Unstructured, error) {
+	logger := logging.FromContext(ctx).Desugar()
+	deploymentName := ra.GetName()
+	logger.Info("Got ra", zap.Any("receiveAdapter", ra))
+	logger.Info("Got ra name "+deploymentName, zap.Any("deploymentName", deploymentName))
+	namespace := src.Namespace
+	name := src.Name
+	gvk := schema.GroupVersionKind{
+		Group:   "keda.k8s.io",
+		Version: "v1alpha1",
+		Kind:    "ScaledObject",
+	}
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	scaledObjectResourceInterface := r.DynamicClientSet.Resource(gvr).Namespace(namespace)
+	if scaledObjectResourceInterface == nil {
+		return nil, fmt.Errorf("unable to create dynamic client for ScaledObject")
+	}
+	scaledObjectUnstr, err := r.generateKedaScaledObjectUnstructured(ctx, ra, src)
+	if err != nil {
+		return nil, err
+	}
+	created, err := scaledObjectResourceInterface.Create(scaledObjectUnstr, metav1.CreateOptions{})
```

Comment:
Wouldn't we want to do the reverse order in a reconciler? Don't we want to first `Get(scaledObjectUnstr...)` and if it doesn't exist, only then actually create it?  If it already exists, then we update it as needed.